src/main/java/org/jabref/gui/exporter/CreateModifyExporterDialogViewModel.java

```diff
@@ -0,0 +1,89 @@
+    private final PreferencesService preferences;
+
+    private final SimpleStringProperty name; //prevent saveExporter from saving this if it's null
+    private final SimpleStringProperty layoutFile;
+    private final SimpleStringProperty extension;
+
+
+    public CreateModifyExporterDialogViewModel(TemplateExporter exporter, DialogService dialogService, PreferencesService preferences) {
+        this.exporter = exporter;
+        this.dialogService = dialogService;
+        this.preferences = preferences;
+        //Set text of each of the boxes
+
+    }
+
+    public TemplateExporter saveExporter() {//void?
+        Path layoutFileDir = Paths.get(layoutFile.get()).getParent();
+        if (layoutFileDir != null) {
+            preferences.put(JabRefPreferences.EXPORT_WORKING_DIRECTORY, layoutFileDir.toString()); //fix to work with PreferencesService
+
+        }
+        preferences.saveExportWorkingDirectory(layoutFileDir.toString()); //See Swing class CustomExportDialog for more on how implement this
+        // Create a new exporter to be returned to ExportCustomizationDialog, which requested it
+        LayoutFormatterPreferences layoutPreferences = preferences.getLayoutFormatterPreferences();
+        SavePreferences savePreferences = preferences.LoadForExportFromPreferences();
+        String filename = layoutFile.get(); //change var name?
+        String extensionString = extension.get(); //change var name?
+        String lfFileName;
+        //You might want to move the next few lines into logic because it also appears in JabRefPreferences
+        if (filename.endsWith(".layout")) {
+            lfFileName = filename.substring(0, filename.length() - ".layout".length());
+        } else {
+            lfFileName = filename;
+        }
+        if (extensionString.contains(".")) {
+            extension.set(extensionString.substring(extensionString.indexOf('.') + 1, extensionString.length()));
+        }
+        FileType fileType = StandardFileType.newFileType(extensionString);
+        TemplateExporter format = new TemplateExporter(name.get(), name.get(), lfFileName, null, fileType, layoutPreferences,
+                                                       savePreferences);
+        format.setCustomExport(true);
+        return format;
+    }
+
+    public String getExportWorkingDirectory() {//i.e. layout dir
+        return preferences.getExportWorkingDirectory();
+    }
+
+    public void browse() {
+        FileDialogConfiguration fileDialogConfiguration = new FileDialogConfiguration.Builder()
+            .addExtensionFilter(Localization.lang("Custom layout file"), StandardFileType.LAYOUT)
+            .withDefaultExtension(Localization.lang("Custom layout file"), StandardFileType.LAYOUT)
+            .withInitialDirectory(getExportWorkingDirectory()).build();
+        dialogService.showFileOpenDialog(fileDialogConfiguration).ifPresent(f -> setText(f.toAbsolutePath().toString())); //implement setting the text
```

Comment:
To set the text, you would create a StringProperty here, e.g. selectedPath and then bind that property to a textfield in the View

Reply:
Fixed it.  I think I had the property, but I hadn't realized this is what I needed here.  Thanks for your recommendation!